presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcDeleteDeltaPageSource.java

```diff
@@ -75,7 +75,8 @@ public OrcDeleteDeltaPageSource(
             String sessionUser,
             Configuration configuration,
             HdfsEnvironment hdfsEnvironment,
-            FileFormatDataSourceStats stats)
+            FileFormatDataSourceStats stats,
+            boolean originalFilesPresent)
```

Comment:
This is not the right name. (And I am not sure about code structure too.)

This is because, when `originalFilesPresent`, this changes behavior of this class, in a incompatible matter.
So, data processing when there are no original files works some way _A_ and when you create one orignal file, this class starts to beheave in a way _B_, which is not compatible with _A_.
For example, it discards transaction id form delta files.

Delete deltas for original files are different. I don't think we should discard data. We should verify there isn't the information we would be discarding.

If my understanding is correct, I suggest you rename originalFilesPresent -> originalFiles
and when set, validate the file does not have those transactionId, etc information we would be ignoring.

Reply:
Delete deltas structure is same for deltas, base as well as original files. For delete deltas, we don't need to read bucket ID as we filter out delete delta with the same bucket as that of original file split (OrcDeletedRows.java -> getDeletedRows). We calculate original file row ID synthetically by reading the ORC footer of all the original files present before in lexicographically order + the location in the current ORC file. 

So, there is no need to read original transaction ID, bucket ID from delta files to save read time.